### DISCUSSION TITLE
Impersonating account

### DISCUSSION THREAD

**Discussion by jpsains** [ethereumjs/ethereumjs-monorepo#2424] [Q&A]
Hey, 

I want to test sending a transaction impersonating a user's address using @ethereumjs/vm and @ethereumjs/statemanager, is this possible?

**Comment by jpsains** [comment]
If not possible, is callstatic available, with from param?

**Reply by acolytec3** [reply]
If I understand your question correctly, I'm assuming you want to leverage mainnet state maybe?  If so, the `ethersStateManager` + `vm.runTx` should do nicely for these purposes.  

You can use the [EthersStateManager](https://github.com/ethereumjs/ethereumjs-monorepo/tree/master/packages/statemanager#ethersstatemanager) to retrieve any mainnet state desired and then just create transactions using our transaction library and then passing those to `vm.runTx()`.  We don't do signature verification in `runTx` (this is assumed to be handled elsewhere), so you can populate any from/to/data you want.  If you need to skip balance checks, just pass in `skipBalanceCheck` to the `RunTxOpts` of `runTx`.
